src/xcode/ENA/ENA/Source/Scenes/Containers/TopBottomContainerViewController.swift

```diff
@@ -75,71 +79,94 @@ class TopBottomContainerViewController<TopViewController: UIViewController, Bott
+					  let animationDuration = notification.userInfo?[UIResponder.keyboardAnimationDurationUserInfoKey] as? Double,
+					  let animationCurve = notification.userInfo?[UIResponder.keyboardAnimationCurveUserInfoKey] as? Int else {
 					return
 				}
-
-				self.bottomViewBottomAnchorConstraint.constant = -(keyboardHeight - self.view.safeAreaInsets.bottom - self.bottomViewController.view.safeAreaInsets.bottom)
-
-				UIView.animate(withDuration: 0.5) {
-					self.view.layoutIfNeeded()
-				}
+				
+				self.bottomViewBottomAnchorAnchorConstraint.constant = -keyboardFrame.height
+				
+				let options = UIView.AnimationOptions(rawValue: ((UInt(animationCurve << 16))))
+				UIView.animate(withDuration: animationDuration, delay: 0, options: options, animations: { [weak self] in
+					self?.view.layoutIfNeeded()
+				}, completion: nil)
 			}
-			.store(in: &subscriptions)
-
+			.store(in: &keyboardSubscriptions)
+		
 		NotificationCenter.default.ocombine.publisher(for: UIApplication.keyboardWillHideNotification)
-			.sink { [weak self] _ in
-				self?.bottomViewBottomAnchorConstraint.constant = 0
-
-				UIView.animate(withDuration: 0.5) {
+			.sink { [weak self] notification in
+				
+				guard let self = self,
+					  let keyboardFrame = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect,
+					  let animationDuration = notification.userInfo?[UIResponder.keyboardAnimationDurationUserInfoKey] as? Double,
+					  let animationCurve = notification.userInfo?[UIResponder.keyboardAnimationCurveUserInfoKey] as? Int else {
+					return
+				}
+				
+				self.bottomViewBottomAnchorAnchorConstraint.constant = -self.view.safeAreaInsets.bottom
+				
+				let options = UIView.AnimationOptions(rawValue: ((UInt(animationCurve << 16))))
+				UIView.animate(withDuration: animationDuration, delay: 0, options: options, animations: { [weak self] in
 					self?.view.layoutIfNeeded()
+				}, completion: nil)
+			}
+			.store(in: &keyboardSubscriptions)
+		
+		NotificationCenter.default.ocombine.publisher(for: UIApplication.keyboardDidHideNotification)
+			.sink { [weak self] notification in
+				guard let self = self,
+					  let keyboardFrame = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect else {
+					return
 				}
+				self.footerViewHandler?.didShowKeyboard(keyboardFrame)
 			}
-			.store(in: &subscriptions)
-
-
-		keyboardDidShownObserver = NotificationCenter.default.addObserver(forName: UIResponder.keyboardDidShowNotification, object: nil, queue: OperationQueue.main) { notification in
-			guard let keyboardSize = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue else {
-				return
+			.store(in: &keyboardSubscriptions)
+		
+		NotificationCenter.default.ocombine.publisher(for: UIApplication.keyboardDidHideNotification)
+			.sink { [weak self] _ in
+				guard let self = self else {
+					return
+				}
+				self.footerViewHandler?.didHideKeyboard()
```

Comment:
I think we can simplify this
```suggestion
				self?.footerViewHandler?.didHideKeyboard()
```

Reply:
Done